projects/vdk-plugins/vdk-data-source-git/src/vdk/plugin/data_source_git/git_source.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import os
+import tempfile
+from typing import Iterator
+from typing import List
+from typing import Optional
+
+from dulwich import porcelain
+from vdk.plugin.data_source_git.utils import detect_language
+from vdk.plugin.data_source_git.utils import is_test_file
+from vdk.plugin.data_sources.config import config_class
+from vdk.plugin.data_sources.config import config_field
+from vdk.plugin.data_sources.data_source import DataSourcePayload
+from vdk.plugin.data_sources.data_source import IDataSource
+from vdk.plugin.data_sources.data_source import (
+    IDataSourceConfiguration,
+)
+from vdk.plugin.data_sources.data_source import IDataSourceStream
+from vdk.plugin.data_sources.factory import data_source
+from vdk.plugin.data_sources.state import IDataSourceState
+
+DESCRIPTION = """Git data source.
+Extract content from Git repositories and associated file metadata.
+"""
+
+
+@config_class(name="git", description=DESCRIPTION)
+class GitDataSourceConfiguration(IDataSourceConfiguration):
+    git_url: str = config_field(description="Git URL that would be cloned. ")
+    git_ssh_key: Optional[str] = config_field(
+        description="SSH key to use when cloning the repo."
+        "Leave empty if no authentication is needed",
+        default="",
+    )
+
+
+@data_source(name="git", config_class=GitDataSourceConfiguration)
+class GitDataSource(IDataSource):
+    """
+    Data source who is only generating some dummy data for testing purposes.
```

Comment:
```suggestion
    Data source implementation is fetching data from git repository and associated metadata.
    For now it provides only 1 stream (for the whole repo). 
    Eventually, it might make sense to allow users to configure streams - maybe branches, or directories or something else. 
```